model/mmwave-sidelink-mac.cc

```diff
@@ -161,22 +166,51 @@ MmWaveSidelinkMac::DoSlotIndication (mmwave::SfnSf timingInfo)
         // discard the tranmission opportunity and go to the next transmission
         continue;
       }
+      if(m_phySapProvider->IsChannelIdle()){
+        // otherwise, forward the packet to the PHY
+        Ptr<PacketBurst> pb = CreateObject<PacketBurst> ();
+        pb->AddPacket (txBuffer->second.front ().pdu);
+        m_phySapProvider->AddTransportBlock (pb, *it);
+        txBuffer->second.pop_front ();
+      }/*else{
+        //se il segnale è per me allora ricevo, sennò sto fermo
+        NS_LOG_INFO ("Prepare for reception from rnti " << it->m_rnti);
+        m_phySapProvider->PrepareForReception (it->m_rnti);
+        //m_phySapProvider->PrepareForReception (m_sfAllocInfo[timingInfo.m_slotNum]);
```

Comment:
Prova a de-commentare questo, passandogli come parametro l'rnti (rnti = ID che identifica il veicolo) dell'altro veicolo, quello con cui voglio comunicare. Puoi provare facendo un po' sporca tipo questa: se il mio rnti è 0, allora chiamo PrepareForReception (1), e viceversa. Se il mio rnti è 2, allora chiamo PrepareForReception (3) , e viceversa. 

Poi sposta questa riga in modo che PrepareForReception venga chiamato quando non ho niente da trasmettere, o quando il canale è occupato